cpp/bridges_module/bridges_test.cpp

```diff
@@ -0,0 +1,213 @@
+#include <queue>
+
+#include <gtest/gtest.h>
+#include <mg_generate.hpp>
+#include <mg_graph.hpp>
+#include <mg_test_utils.hpp>
+
+#include "algorithm/bridges.hpp"
+
+namespace {
+
+std::uint64_t CountComponents(const mg_graph::Graph<> &graph) {
+  std::uint64_t n_components = 0;
+  std::vector<bool> visited(graph.Nodes().size());
+
+  for (auto &node : graph.Nodes()) {
+    if (visited[node.id]) continue;
+    ++n_components;
+
+    std::queue<std::uint64_t> queue;
+    visited[node.id] = true;
```

Comment:
If the queue contains node ids, which are defined as TSize, then it should be of type whichever is Graph, something like this:
```suggestion
template <typename TSize = std::uint64_t>
std::uint64_t CountComponents(const mg_graph::Graph<TSize> &graph) {
  std::uint64_t n_components = 0;
  std::vector<bool> visited(graph.Nodes().size());

  for (auto &node : graph.Nodes()) {
    if (visited[node.id]) continue;
    ++n_components;

    std::queue<TSize> queue;
    visited[node.id] = true;
```

Reply:
That's a nice catch!
But these functions are restricted only to this TU, so you don't need the flexibility of the templates that much. Every created `Graph` here uses the default argument, and it's okay to have a function that only accepts that.
If he tried to call this function with any other TSize, it would be a compilation error, so it wouldn't go undetected.
This boils down to not overcomplicate things if not necessary, in C++ it's mostly trying not to use templates if you can. :D

Also, because Graph objects of different template arguments cannot be converted to each other, this showcases nicely how you don't need the default argument here.
```
template <typename TSize>
std::uint64_t CountComponents(const mg_graph::Graph<TSize> &graph)
```
`TSize` will be deduced from the graph type. 
If you call this function with a `Graph<>` object, the TSize will be the default type of the Graph template argument.